src/type/validate.js

```diff
@@ -438,7 +438,20 @@ function validateUnionMembers(
   }
 
   const includedTypeNames = Object.create(null);
+  const includedInvalidTypes = Object.create(null);
```

Comment:
I think it strange behavior where after fixing one error you would get the same error in the next location.
We should definitely prevent implicit cascading errors but I think we should never mask explicit errors made by the user.
Also if we add this error reduction trick we need to add it in many other places (e.g. if you implement non-interface types) for consistency.

So I think moving check to beginning of the loop and adding `continue` would be enough.

Reply:
Note that this does not hide any errors - there is only one message, but all locations are reported and the same message applies to them all.